Issue when loading detection.so

## issue below
peter@ubuntu:\~/container_cpu_detection$ make
gcc -std=c99 -Wall -shared -g -fPIC detection.c -o detection.so
gcc sysconf_test.c -o sysconf_test
peter@ubuntu:~/container_cpu_detection$ make test
docker run -ti --rm --cpuset-cpus 0,1 --cpu-quota 200000 -v /home/peter/container_cpu_detection/detection.so:/usr/lib/detection.so -v /home/peter/container_cpu_detection/sysconf_test:/tmp/sysconf_test -e DETECTION_TARGETS=sysconf_test -e LD_PRELOAD=/usr/lib/detection.so ubuntu /tmp/sysconf_test
/tmp/sysconf_test: symbol lookup error: /usr/lib/detection.so: undefined symbol: dlsym
Makefile:20: recipe for target 'test' failed
make: *** [test] Error 127

## my environment

gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.4.0-1ubuntu1\~18.04.1' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.4.0 (Ubuntu 7.4.0-1ubuntu1\~18.04.1)

--- a/.gitignore
+++ b/.gitignore
@@ -1,1 +1,2 @@
 *.so
+sysconf_test

--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,8 @@ DOCKER_RUN_OPTS = $(shell echo "-ti --rm" \
 					"-e LD_PRELOAD=/usr/lib/detection.so")
 
 $(versioned_so): detection.c
-	gcc -std=c99 -Wall -shared -g -fPIC -ldl detection.c -o $(versioned_so)
+	gcc -std=c99 -Wall -shared -g -fPIC -Wl,--no-as-needed -ldl detection.c -o $(versioned_so)
+
 	gcc sysconf_test.c -o sysconf_test
 
 clean:

